packages/pwa/app/commerce-api/index.js

```diff
@@ -10,9 +10,36 @@ import * as sdk from 'commerce-sdk-isomorphic'
 import {getAppOrigin} from 'pwa-kit-react-sdk/utils/url'
 import OcapiShopperBaskets from './ocapi-shopper-baskets'
 import OcapiShopperOrders from './ocapi-shopper-orders'
-import {getTenantId, isError, isTokenValid} from './utils'
+import {getTenantId, isServer, isError, isTokenValid} from './utils'
 import Auth from './auth'
 import EinsteinAPI from './einstein'
+import http from 'http'
+import https from 'https'
+
+const KEEP_ALIVE_MS = 8000
+
+const HTTP_AGENT = isServer
+    ? new http.Agent({
+          keepAlive: true,
+          keepAliveMsecs: KEEP_ALIVE_MS
+      })
+    : undefined
+const HTTPS_AGENT = isServer
+    ? new https.Agent({
+          keepAlive: true,
+          keepAliveMsecs: KEEP_ALIVE_MS
+      })
+    : undefined
+
+/**
+ * Node.js http or https Agent with keepAlive support to re-use Commerce API
+ * connections server-side.
+ * @param url
+ * @returns {module:http.Agent|module:https.Agent}
+ */
+function getAgent(url) {
+    return url.protocol === 'http:' ? HTTP_AGENT : HTTPS_AGENT
+}
```

Comment:
Open question. Do you think that we should be adding this functionality into the CommerceApi class itself, or providing the agent via the classes `config` argument, similarly to how each sdk class is instanciated?

This would suggest that we do something in the `_app-config.js` like: 

```
AppConfig.restore = (locals = {}) => {
    const agent = getAgent
    locals.api = new CommerceAPI({...apiConfig, locale, currency, agent})
}
```

Reply:
Good question. It makes sense to keep all the configurations in the same place. I made the change. It looks cleaner that way.